test/DateRangeTest.java

```diff
@@ -0,0 +1,49 @@
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+import sdk.annotations.Attribute;
+import sdk.converter.ObjectConverter;
+import sdk.converter.ParserContext;
+import sdk.data.DataSetItem;
+import sdk.models.DateRange;
+import sdk.models.DateTimeRange;
+
+/**
+ * Created by Orozco on 9/19/17.
+ */
+public class DateRangeTest {
+
+
+    class DateRangeTestClass {
+        @Attribute(index = 0)
+        private DateTime createDate;
+
+        public DateTime getCreateDate() {
+            return createDate;
+        }
+
+        public void setCreateDate(DateTime createDate) {
+            this.createDate = createDate;
+        }
+    }
+
+
+    @Test
+    public void testDateRange() {
```

Comment:
This test is valid but it doesn't cover all the things I'd like. I think we should have two more tests where you include some mocked JSON. One JSON payload should be for a data set item that has been sent with a date range. The second where it is sent as a date. 

You should parse that in and run it through the parser and make sure that getDate /getDateTime and getDateRange/getDateTimeRange all function as expected.